py/server/deephaven/column.py

```diff
@@ -32,47 +32,122 @@ def __repr__(self):
         return self.name
 
 
-@dataclass
-class Column:
-    """ A Column object represents a column definition in a Deephaven Table. """
-    name: str
-    data_type: DType
-    component_type: DType = None
-    column_type: ColumnType = ColumnType.NORMAL
+class ColumnDefinition(JObjectWrapper):
+    """A Deephaven column definition."""
+
+    j_object_type = _JColumnDefinition
+
+    @staticmethod
+    def of(
+            name: str,
+            data_type: DType,
+            component_type: DType = None,
+            column_type: ColumnType = ColumnType.NORMAL,
+    ) -> 'ColumnDefinition':
+        return ColumnDefinition(
+            _JColumnDefinition.fromGenericType(
+                name,
+                data_type.j_type.jclass
+                if hasattr(data_type.j_type, "jclass")
+                else data_type.qst_type.clazz(),
+                component_type.qst_type.clazz() if component_type else None,
+                column_type.value,
+            )
+        )
+
+    def __init__(self, j_column_definition: jpy.JType):
+        self.j_column_definition = j_column_definition
 
     @property
-    def j_column_header(self):
-        return _JColumnHeader.of(self.name, self.data_type.qst_type)
-
-    @property
-    def j_column_definition(self):
-        if hasattr(self.data_type.j_type, 'jclass'):
-            j_data_type = self.data_type.j_type.jclass
-        else:
-            j_data_type = self.data_type.qst_type.clazz()
-        j_component_type = self.component_type.qst_type.clazz() if self.component_type else None
-        j_column_type = self.column_type.value
-        return _JColumnDefinition.fromGenericType(self.name, j_data_type, j_component_type, j_column_type)
-
-
-@dataclass
-class InputColumn(Column):
-    """ An InputColumn represents a user defined column with some input data. """
-    input_data: Any = field(default=None)
-
-    def __post_init__(self):
+    def j_object(self) -> jpy.JType:
+        return self.j_column_definition
+
+    @cached_property
+    def name(self) -> str:
+        return self.j_column_definition.getName()
+
+    @cached_property
+    def data_type(self) -> DType:
+        return from_jtype(self.j_column_definition.getDataType())
+
+    @cached_property
+    def component_type(self) -> DType:
+        return from_jtype(self.j_column_definition.getComponentType())
+
+    @cached_property
+    def column_type(self) -> ColumnType:
+        return ColumnType(self.j_column_definition.getColumnType())
+
+
+class Column(ColumnDefinition):
```

Comment:
Is `Column` needed anymore?

Reply:
It would be a breaking change to remove, as users may be relying on the Column constructor, for example

```python
some_method([Column("Foo", dtypes.int32), Column("Bar", dtypes.str)])
```

Reply:
https://github.com/deephaven/deephaven-core/issues/5940